magnetapiclient.test.js

```diff
@@ -0,0 +1,176 @@
+const expect = require('expect');
+const nock = require('nock'); 
+
+var MagnetAPIClient = require('./magnetapiclient');
+
+
+
+describe('MagnetAPIClient', function () {
+
+    const ENDPOINT_URI = "http://magnetapi.klangoo.com/NewsAgencyService.svc";
+    const CALK = "ENTER_YOUR_CALK";
+    const SECRET_KEY = "ENTER_YOUR_SECRET_KEY";
+    const articleUID = "2017/03/28/example-api-usage";
+
+    let client;
+    let api;
+
+    beforeEach(function () {
+
+        client = new MagnetAPIClient(ENDPOINT_URI, CALK, SECRET_KEY);
+        api = nock(ENDPOINT_URI, {
+            reqheaders: {
+                'content-type': 'application/x-www-form-urlencoded'
+            }
+        });
+    });
+
+    it('should be able to add an article', function (done) {
+
+        const request = {
+            text: 'SAMPLE ARTICLE TEXT',
+            title: 'SAMPLE ARTICLE TITLE',
+            insertDate: '23 JAN 2017 10:12:00 +01:00',
+            url: 'http://demo.klangoo.com/article-demo/api-example',
+            articleUID,
+            source: 'klangoo.com',
+            language: 'en',
+            format: 'json'
+        };
+
+        api.post('/AddArticle', function (body) {
+
+            return body.calk === CALK &&
+                   body.articleUID === request.articleUID &&
+                   body.format === request.format &&
+                   body.language === request.language &&
+                   body.source === request.source &&
+                   body.text === request.text &&
+                   body.title === request.title &&
+                   body.url === request.url &&
+                   typeof body.timestamp !== 'undefined' &&
+                   typeof body.signature !== 'undefined';
+        }).reply(200, { status: 'OK' });
+
+
+        client.CallWebMethod('AddArticle', request, 'POST', function (data) {
+
+            const parsedResult = JSON.parse(data);
+
+            expect(parsedResult.status).toEqual('OK');
+            done();
+        });
+    });
+
+    it('should be able to update an article', function (done) {
+
+        const request = {
+            text: 'SAMPLE ARTICLE TEXT UPDATED',
+            updateDate: '24 JAN 2017 10:12:00 +01:00',
+            articleUID,
```

Comment:
seems to be missing value for this key

Reply:
It is just es6 shorthand for
articleUID: articleUID,